sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -157,6 +149,30 @@ private static List<ByteBuffer> serialize(List<TelemetryItem> telemetryItems) {
         }
     }
 
+    private static String toJson(List<TelemetryItem> telemetryItems) throws IOException {
+        try (StringWriter debug = new StringWriter()) {
+            for (int i = 0; i < telemetryItems.size(); i++) {
+                JsonWriter jsonWriter = JsonProviders.createWriter(debug);
+                TelemetryItem telemetryItem = telemetryItems.get(i);
+                telemetryItem.toJson(jsonWriter);
+                jsonWriter.flush();
+                if (i < telemetryItems.size() - 1) {
+                    debug.write('\n');
+                }
+            }
+            return debug.toString();
+        }
+    }
+
+    private static String removeIKeys(List<TelemetryItem> telemetryItems, String json) {
+        Set<String> iKeys
+            = telemetryItems.stream().map(TelemetryItem::getInstrumentationKey).collect(Collectors.toSet());
+        for (String instrumentationKey : iKeys) {
+            json = json.replace(instrumentationKey, "***");
+        }
+        return json;
+    }
```

Comment:
For each `TelemetryItem` here is it effectively immutable at the point this method is called? If it is, might I suggest mutating each to set `instrumenetationKey` to `***` before generating JSON and resetting the value back to what it was before completing JSON generation?

If this is something called in a hot path or with a lot of items this could be a large amount of processing. Ex, we have 50 items and the generated JSON is 256 KB, this will need to process through 12.5 MB of String with all the allocation overhead for that single call.

Reply:
From telemetry data, a TelemetryItem as JSON is more around (or less than) 1 KB.

In addition, the processing is done if the user enables the log at the verbose level:
https://github.com/Azure/azure-sdk-for-java/blob/474365cedfd5093e9e84cc1952fe1db04ea027e1/sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/pipeline/TelemetryItemExporter.java#L134

We could perhaps in this case avoid extra complexity in the code.